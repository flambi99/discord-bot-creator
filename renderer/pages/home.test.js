import { render, fireEvent, screen } from '../../test_setup/test-utils';
import React from 'react'
import Dashboard from './home'
import renderer from "react-test-renderer"


it("check button render", () => {
    const { queryByTestId } = render(<Dashboard/>)

    const createBotBtn = queryByTestId("create-bot")
    const addBotBtn = queryByTestId("add-bot")

    expect(createBotBtn).toBeTruthy()
    expect(addBotBtn).toBeTruthy()
})

it("buttons with correct name", () => {
    const { queryByTestId } = render(<Dashboard/>)

    const createBotBtn = queryByTestId("create-bot")
    const addBotBtn = queryByTestId("add-bot")

    expect(createBotBtn.textContent).toBe("Create new bot")
    expect(addBotBtn.textContent).toBe("Add Bot")
})

it("matches snapshot", () => {
    const tree = renderer.create(<Dashboard/>).toJSON()

    expect(tree).toMatchSnapshot()
})

