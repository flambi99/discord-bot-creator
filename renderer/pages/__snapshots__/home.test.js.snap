// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="mt-4 container"
>
  <div
    className="align-items-stretch row"
  >
    <form
      className=""
    >
      <button
        className="mx-3 mt-3 btn btn-primary"
        data-testid="create-bot"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        Create new bot
      </button>
      <button
        className="mt-3 btn btn-primary"
        data-testid="add-bot"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Add Bot
      </button>
    </form>
  </div>
</div>
`;
